sentry_sdk/integrations/fastapi.py

```diff
@@ -0,0 +1,122 @@
+from sentry_sdk._types import MYPY
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import DidNotEnable
+from sentry_sdk.integrations.starlette import (
+    SentryStarletteMiddleware,
+    StarletteIntegration,
+)
+from sentry_sdk.tracing import SOURCE_FOR_STYLE, TRANSACTION_SOURCE_ROUTE
+from sentry_sdk.utils import transaction_from_function
+
+if MYPY:
+    from typing import Any, Callable, Dict
+
+    from sentry_sdk._types import Event
+
+try:
+    from fastapi.applications import FastAPI
+    from fastapi.requests import Request
+except ImportError:
+    raise DidNotEnable("FastAPI is not installed")
+
+try:
+    from starlette.types import ASGIApp, Receive, Scope, Send
+except ImportError:
+    raise DidNotEnable("Starlette is not installed")
+
+
+_DEFAULT_TRANSACTION_NAME = "generic FastApi request"
+
+
+class FastApiIntegration(StarletteIntegration):
+    identifier = "fastapi"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        StarletteIntegration.setup_once()
+        patch_middlewares()
+
+
+def patch_middlewares():
+    # type: () -> None
+
+    old_build_middleware_stack = FastAPI.build_middleware_stack
```

Comment:
The Starlette base middleware will probably change to fix a couple of issues, and this logic to build the middleware stack will probably change in the future. But I would expect the new version should be easier to patch too.

Reply:
Good to know. (And yes, it was a bit cumbersome to instrument the middlewares :-) )
Any plans on when this will change?

Reply:
Not sure yet because it will probably have to be fixed in Starlette. There are several people trying to solve it, I still have to help review all that too. And then port/update it in FastAPI

Reply:
> The Starlette base middleware will probably change to fix a couple of issues

Which issues?